azure-pipelines.yml

```diff
@@ -81,8 +81,42 @@ stages:
               pyupgrade --py3-plus || EXIT_STATUS=$?
               exit $EXIT_STATUS
 
-  - stage: required
+  - stage: import_ge
     dependsOn: [lint]
+    pool:
+      vmImage: 'ubuntu-18.04'
+    jobs:
+      - job: import_ge
+
+        strategy:
+         matrix:
+           Python37:
+             python.version: '3.7'
+           Python38:
+             python.version: '3.8'
+           Python39:
+             python.version: '3.9'
+
+
+        steps:
+         - task: UsePythonVersion@0
+           inputs:
+             versionSpec: '$(python.version)'
+           displayName: 'Use Python $(python.version)'
+
+         - bash: python -m pip install --upgrade pip==21.3.1
+           displayName: 'Update pip'
+
+         - script: |
+             pip install  .
```

Comment:
Won't this install anything that is imported in the codebase rather than `requirements.txt`?

Reply:
it'll use the `setup.py` to download a local installation so I think this is okay?

Reply:
https://github.com/great-expectations/great_expectations/blob/develop/setup.py#L37-L39 shows that it is equivalent to `requirements.txt`

Reply:
👀 Checking

Reply:
(It did not install sqlalchemy locally but checking in-pipeline and in the pip docs again)

Reply:
Looks like we are good, using pip to install from a directory e.g. `pip install .` makes pip look for a setup.py file (https://pip.pypa.io/en/stable/cli/pip_install/#argument-handling) which in our case specifies requirements.txt as our requirements and other requirements as `extras_require`.

Reply:
Some research confirms what you say @cdkini and `setup.py` does not allow `sqlalchemy` unless you use the "extra" e.g. `great_expectations[sqlalchemy]`